### PR TITLE
Add clang-tidy to autoformat C/C++

### LINKED ISSUE #128: Add clang tidy
**Acceptance Criteria**
- C/C++ uses CamelCase
- Fortran uses snake case

**Ideas**

### PR DESCRIPTION
**Pull request by boulderdaze**
Closes #128
- Applied clang-tidy to C/C++ code
- Modified `TUVX *CreateTuvx(const char *config_path, int *error_code)` to verify the existence of configuration file before creating TUVX object to prevent memory leak when `TUVX` creates `YAML` node.  
-  `Dockerfile.fortran-gcc failure` is caused by `call test_tuvx_fort_api()` and my understanding is that Kyle is working on TUVX
```
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: possible
   fun:malloc
   fun:__tuvx_cross_section_no2_tint_MOD_constructor
   fun:__tuvx_cross_section_factory_MOD_cross_section_builder
   fun:__tuvx_photolysis_rates_MOD_add
   fun:__tuvx_photolysis_rates_MOD_constructor
   fun:__tuvx_core_MOD_constructor
   fun:InternalCreateTuvx
   fun:_ZN6musica4TUVX6CreateERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
   fun:CreateTuvx
   fun:__musica_tuvx_MOD_constructor
   fun:test_tuvx_fort_api.1
   fun:main
}
==397== LEAK SUMMARY:
==397==    definitely lost: 0 bytes in 0 blocks
==397==    indirectly lost: 1,607,111 bytes in 875 blocks
==397==      possibly lost: 1,632 bytes in 1 blocks
==397==    still reachable: 95 bytes in 4 blocks
==397==         suppressed: 16 bytes in 2 blocks
```
- Intel - Kyle's open PR will fix this
- Nvhpc - a separate issue will be created
- mac/gcc-13 - uncertain why it fails

### CODE REVIEW

File: docker/Dockerfile.fortran-nvhpc [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #
 FROM nvcr.io/nvidia/nvhpc:23.7-devel-cuda12.2-ubuntu22.04
 
-ARG MUSICA_GIT_TAG=main
+ARG MUSICA_GIT_TAG=801807d
```
Comment (K20shores) [inline docker/Dockerfile.fortran-nvhpc:9]:
```suggestion
ARG MUSICA_GIT_TAG=main
```

In the future, you can use `docker build -t musica --build-arg="MUSICA_GIT_TAG=801807d"`

File: fortran/test/fetch_content_integration/CMakeLists.txt [hunk 2]
```diff
@@ -9,7 +9,7 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 
 include(FetchContent)
 
-set(MUSICA_GIT_TAG "main" CACHE STRING "Git tag for the musica_fortran repository")
+set(MUSICA_GIT_TAG "801807d" CACHE STRING "Git tag for the musica_fortran repository")
```
Comment (K20shores) [inline fortran/test/fetch_content_integration/CMakeLists.txt:12]:
```suggestion
set(MUSICA_GIT_TAG "main" CACHE STRING "Git tag for the musica_fortran repository")
```

**Review by K20shores — CHANGES_REQUESTED** [review]

File: docker/Dockerfile.fortran-nvhpc (hunk 1 above)
Comment (boulderdaze) [reply docker/Dockerfile.fortran-nvhpc]: That's good to know

File: fortran/test/fetch_content_integration/CMakeLists.txt (hunk 2 above)
Comment (boulderdaze) [reply fortran/test/fetch_content_integration/CMakeLists.txt]: I changed it back to `main`. I guess we don't worry about failing the test for fetch content if it always sets to `main`?

Comment (K20shores) [reply fortran/test/fetch_content_integration/CMakeLists.txt]: The CI uses the current branch name for the tests when you make a PR

**Review by mattldawson — APPROVED** [review]
looks great!

**Review by K20shores — APPROVED** [review]

**Pull request merged**